sdk/keyvault/azure_security_keyvault_keys/benches/benchmarks.rs

```diff
@@ -77,11 +81,171 @@ fn key_operations_benchmark(c: &mut Criterion) {
     });
 }
 
+fn key_2901_benchmark_default(c: &mut Criterion) {
+    const ENV_NAME: &str = "AZURE_KEYVAULT_URL";
+
+    // Check if the environment variable is set thus allowing the benchmarks to run
+    if std::env::var(ENV_NAME).is_err() {
+        println!("Skipping benchmarks. Set {} to run.", ENV_NAME);
+        return;
+    }
+    const KEY_NAME: &str = "key-2901";
+
+    //    let rt = tokio::runtime::Runtime::new().unwrap();
+    let rt = tokio::runtime::Builder::new_current_thread()
+        .enable_all()
+        .build()
+        .unwrap();
+
+    // Setup the KeyClient and the CreateKeyParameters
+    async fn setup_key_client() -> (KeyClient, CreateKeyParameters) {
+        let keyvault_url: String =
+            std::env::var(ENV_NAME).unwrap_or_else(|e| panic!("{} not set: {}", ENV_NAME, e));
+        let credential = credentials::from_env(None).unwrap();
+        let client: KeyClient = KeyClient::new(&keyvault_url, credential.clone(), None).unwrap();
+        let body = CreateKeyParameters {
+            kty: Some(KeyType::RSA),
+            key_size: Some(2048),
+            ..Default::default()
+        };
+        (client, body)
+    }
+
+    let (client, body) = rt.block_on(async { setup_key_client().await });
+
+    // Create a key with name
+    async fn create_key(
+        name: &str,
+        client: &KeyClient,
+        body: CreateKeyParameters,
+    ) -> Result<Key, azure_core::Error> {
+        // Create a new key version
+        client
+            .create_key(name, body.try_into()?, None)
+            .await?
+            .into_body()
+            .await
+    }
+
+    // prep key in order to run the benchmark in a clean state
+    let _ = rt.block_on(async { create_key(KEY_NAME, &client, body.clone()).await });
+
+    let dek = rand::random::<[u8; 32]>(); // Generate a random 32-byte data encryption key (DEK)
+
+    c.bench_function("key_2901_benchmark_default", |b| {
+        b.to_async(&rt).iter(|| async {
+            let parameters = KeyOperationParameters {
+                algorithm: Some(EncryptionAlgorithm::RsaOAEP256),
+                value: Some(dek.to_vec()),
+                ..Default::default()
+            };
```

Comment:
Might want to do this outside the benchmark and clone it as needed. The serialization will take a bit of time as well. No reason to keep doing it. IIRC, the perf repro wasn't doing so, though I could be mistaken.

Reply:
The code from the customer was doing that - I literally cut and pasted his code in and then updated it appropriately. But it doesn't actually matter at the scale of these operations.